CRM/ACL/BAO/ACL.php

```diff
@@ -433,16 +433,21 @@ protected static function matchType($type, $operation) {
    * Delete ACL records.
    *
    * @param int $aclId
-   *   ID of the ACL record to be deleted.
-   *
+   * @deprecated
    */
   public static function del($aclId) {
-    // delete all entries from the acl cache
-    CRM_ACL_BAO_Cache::resetCache();
+    self::deleteRecord(['id' => $aclId]);
+  }
 
-    $acl = new CRM_ACL_DAO_ACL();
-    $acl->id = $aclId;
-    $acl->delete();
+  /**
+   * Event fired before an action is taken on an ACL record.
+   * @param \Civi\Core\Event\PreEvent $event
+   */
+  public static function self_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
+    // Reset cache when deleting an ACL record
+    if ($event->action === 'delete') {
+      CRM_ACL_BAO_Cache::resetCache();
```

Comment:
This is actually wrong - in that it is not cleaning up the deleted acl - unless the cache is already stale. I did look at cleaning it up but it is not quite as simple as switching to 

```
   CRM_Core_DAO::executeQuery('DELETE FROM civicrm_acl_cache WHERE acl_id = ' . (int) $event->id);
   
```

Reply:
@eileenmcnaughton what if we switch this to a post hook?

Reply:
@colemanw so the issue is that the clearing of the contact cache part is tied to the acl cache - if we did 

https://github.com/civicrm/civicrm-core/pull/21219 then we could fix it up to delete just the one acl & then clear the contact cache part